sct/main.go

```diff
@@ -69,7 +107,25 @@ func main() {
 			log.Println("Launched background process... pid", cmd.Process.Pid)
 			os.Exit(0)
 		} else {
-			monitorTime()
+			switch *mode {
+			case "timed":
```

Comment:
Thanks for making the refactors. However, this still has the issue that it'll fork to the background and silently die if there's bad values. You should move the time parsing code to before the if/else statement (L58).